src/buildings/warehouse/warehouse_building.gd

```diff
@@ -0,0 +1,128 @@
+extends Building
+class_name WarehouseBuilding
+
+var specialized_type: EnumAutoload.ResourceType = EnumAutoload.ResourceType.NONE
+
+const SPEC_STORAGE_MULTIPLIER = 4
+const SPEC_PROD_BONUS = 1
+const BASE_PROD_BONUS = 0.2
+
+func _ready():
+	super()
+	data.type = EnumAutoload.BuildingType.STORAGE
+
+
+func apply_upgrades():
+	if EnumAutoload.UpgradeId.CONSTRUCTION_LOGIC_ADV_LOGISTIC in ResourceManager.current_upgrades:
+			get_node("Range").scale = Vector2(1.5, 1.5)
+	# Wait 2 frame to make sure all Area2D changes are setup correctly
+	await get_tree().physics_frame
+	await get_tree().physics_frame
+	check_for_adjacency_multiplier(data.type)
+
+func set_specialisation(_type: EnumAutoload.ResourceType):
+	specialized_type = _type
+	print("Changed")
+	match specialized_type:
+		EnumAutoload.ResourceType.AIR:
+			sprite.self_modulate = Color(1, 0.8, 0.8)
```

Comment:
We could remove this `match` statement if you store those Color values in an array and index it by `specialized_type` directly, as long as the array of Colors follows the indexing of the ResourceType enum.